File: dx-web/src/consts/game-mode.ts
export const GAME_MODES = {
  WORD_SENTENCE: "word-sentence",
  VOCAB_BATTLE: "vocab-battle",
  VOCAB_MATCH: "vocab-match",
  VOCAB_ELIMINATION: "vocab-elimination",
} as const;

export type GameMode = (typeof GAME_MODES)[keyof typeof GAME_MODES];

export const GAME_MODE_LABELS: Record<GameMode, string> = {
  "word-sentence": "连词成句",
  "vocab-battle": "词汇对轰",
  "vocab-match": "词汇配对",
  "vocab-elimination": "词汇消消乐",
};

export const GAME_MODE_OPTIONS: { value: GameMode; label: string }[] =
  Object.entries(GAME_MODE_LABELS).map(([value, label]) => ({
    value: value as GameMode,
    label,
  }));

File: dx-api/app/consts/game_mode.go
package consts

// Game mode values.
const (
	GameModeWordSentence     = "word-sentence"
	GameModeVocabBattle      = "vocab-battle"
	GameModeVocabMatch       = "vocab-match"
	GameModeVocabElimination = "vocab-elimination"
)

// GameModeLabels maps each mode to its Chinese label.
var GameModeLabels = map[string]string{
	GameModeWordSentence:     "连词成句",
	GameModeVocabBattle:      "词汇对轰",
	GameModeVocabMatch:       "词汇配对",
	GameModeVocabElimination: "词汇消消乐",
}

// GameModeOption represents a selectable game mode.
type GameModeOption struct {
	Value string
	Label string
}

// GameModeOptions returns all game modes as an ordered slice.
func GameModeOptions() []GameModeOption {
	return []GameModeOption{
		{Value: GameModeWordSentence, Label: "连词成句"},
		{Value: GameModeVocabBattle, Label: "词汇对轰"},
		{Value: GameModeVocabMatch, Label: "词汇配对"},
		{Value: GameModeVocabElimination, Label: "词汇消消乐"},
	}
}
